api/src/response.rs

```diff
@@ -13,6 +13,19 @@ pub fn okay_with_json(json: &impl Serialize) -> WithStatus<Json> {
     warp::reply::with_status(warp::reply::json(json), StatusCode::OK)
 }
 
+/// `204` No content
+///
+/// The server successfully processed the request, and is not returning any content.
+pub fn no_content() -> WithStatus<Json> {
+    warp::reply::with_status(
+        warp::reply::json(&ErrorMessage {
+            code: 204,
+            msg: "NO_CONTENT",
+        }),
+        StatusCode::NO_CONTENT,
+    )
+}
```

Comment:
In which situation would you need that?

Reply:
I'd like to use it when a book is not available, thereby saying that the request was processed successfully, but it's not returning anything.
So far we said to use `404` in that situation, but that is a bit misleading, because `warp` also returns `404` if you try to access a non-existent route.

Reply:
With "not available" you mean that a book id does not exist? In this case it would be a error 404. I get that it can be confusing. Though, I think the frontend has this requests hard-coded and this would make them ambiguous. The 204 is usually returned when you do an operation which does not return data, e.g. PUT requests.

Reply:
Okay. I will go with 404, and a custom message. What do you think about `BOOK_ID_NOT_FOUND`?

Reply:
Yes, you can do this. If so, we should stick to this convention for the whole system and we have to communicate also have to prepare a list of those. Some of them need to be understood by the frontend code.